src/components/body/slideshow/index.js

```diff
@@ -598,30 +585,45 @@ export default class Slideshow extends PureComponent {
 
   render() {
     const { curSlideIndex, isSliding, translateXUint } = this.state
+    const images = _.get(this.props, 'data.content', [])
+    const total = images.length
+
+    // For slicing images array easily later,
+    // add last `slidesOffset` elements into top of images array.
+    // add first `slidesOffset` elements into bottom of images array.
+    // EX:
+    // slidesOffset: 2
+    // input images: [ a, b, c, d ]
+    // output images: [c, d, a, b, c, d, a, b]
+    const imagesForSlicing = [].concat(
```

Comment:
We can consider use [memoization](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization) to cache the computed results that is not likely to change in `render`

Reply:
It's a simple array composition.
 I would like concat the array in every render since it won't cause any performance issue.

Reply:
After thinking, I will try to use `this.`(global variable) to store the concated images.

Reply:
After thinking twice(we are going to use `memoize-one` in `audio` component), I would try to use `memoize-one` to cache the computed results(`imageForSlicing`).